tests/src/Interop/WinRT/NativeComponent/Component.Contracts.NullableTesting.cpp

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+#include "pch.h"
+#include "Component.Contracts.NullableTesting.h"
+
+namespace winrt::Component::Contracts::implementation
+{
+    bool NullableTesting::IsNull(Windows::Foundation::IReference<int32_t> const& value)
+    {
+        return value == nullptr;
```

Comment:
This is a bit confusing to me. Does the `IReference<T>` have an overload for `==`? Refs aren't initialized to `nullptr` and can't be declared to be `nullptr`, so how is this occurring?

Reply:
C++/WinRT's `IUnknown` has an implicit constructor taking a `nullptr_t` and an overload for comparing two `IUnknown`s. A `null`-valued .NET `Nullable<T>` translates to a null `Windows::Foundation::IReference<T>`, which will compare as equal to `nullptr`.